### PR TITLE
fix openMF#1665: Filter error in AccountsFragment.kt fixed

### LINKED ISSUE #1665: Filter error in Account Fragment
**Summary:**
When user opens Accounts tab and use filter for savings account then error layout does not disappear when filter options are changed. Also if  filters are used in loan accounts, they don't work at all. Also in both if no checkbox is checked then error layout is visible.

https://user-images.githubusercontent.com/70195106/103333078-99be5380-4a92-11eb-9b8a-5b3528703892.mp4

https://user-images.githubusercontent.com/70195106/103333185-fc175400-4a92-11eb-8b29-67c3730588c4.mp4

**Steps to reproduce:**
1 Open Accounts Tab from Home
2 Apply filters for Savings and Loan Accounts

**Expected behaviour:**
Error layout should be updated in savings account filters and List of loan accounts should be updated when filters are applied.

**Device and Android version:**
Lenovo Tab 7
Android 7

### PR DESCRIPTION
**Pull request by kmanikanta335**
Fixes #1665 

Please Add Screenshots If there are any UI changes.

https://github.com/openMF/mifos-mobile/assets/118070186/62fce46e-9128-4413-8cba-37baa9d62d48

Please make sure these boxes are checked before submitting your pull request - thanks!

- ✅  Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- ✅  Run the unit tests with `./gradlew check` to make sure you didn't break anything

- ✅  If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by kmanikanta335** [comment]
@PratyushSingh07 pls review it

File: app/src/main/java/org/mifos/mobile/ui/fragments/AccountsFragment.kt [hunk 1]
```diff
@@ -442,6 +450,10 @@ class AccountsFragment : BaseFragment(), OnRefreshListener {
         if (filteredSavings.size == 0) {
             showEmptyAccounts(getString(R.string.no_saving_account))
         } else {
+            sweetUIErrorHandler?.hideSweetErrorLayoutUI(
+                binding.rvAccounts,
+                binding.layoutError.root,
+            )
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/fragments/AccountsFragment.kt]: encapsulate these lines of code in a function then use them in the code

**Comment by PratyushSingh07** [comment]
Also , I have a doubt . If i click on the filter button without selecting an option should there be any UI layout? I mean if i haven't selected any filter then ideally i should see the screen as it is. Why will i need a UI layout with the message "No Accounts Found" ?

**Comment by kmanikanta335** [comment]
@PratyushSingh07 yes you are right
but there is "cancel" button , when it is pressed then the layout remains as it is.
Basically error layout is just to show that there are no options selected.

https://github.com/openMF/mifos-mobile/assets/118070186/352df1b8-fa8b-42c0-bfbc-112f8f3b4689

By the way I am Manikanta from Dayananda Sagar college of Engineering (5th sem)
I too want crack G'SoC 2024 like you cracked G'SoC 2023 
can you please tell me will Mifos-Mobile take part in gsoc 2024

**Comment by PratyushSingh07** [comment]
If i click on filter button without selecting any option then seeing the screen as it is should be a better alternative. 
@kmanikanta335 you can not predict if mifos mobile or any other project will participate in GSoC 24. It depends on priority of the project and also a lot of other factors.

Also, do not resolve the conversation . let the reviewer do it

**Comment by kmanikanta335** [comment]
@PratyushSingh07 I have made screen remain same after selecting no options in filter
pls review it

https://github.com/openMF/mifos-mobile/assets/118070186/b719d3e6-a15f-4c87-a9dd-96a0fce92407

**Comment by kmanikanta335** [comment]
@therajanmaurya pls merge this too

**Comment by PratyushSingh07** [comment]
LGTM we can merge this @therajanmaurya

**Pull request merged**